src/main/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/BaseRecalibratorDataflow2.java

```diff
@@ -0,0 +1,283 @@
+
+            if (null==baseRecalibrationKnownVariants || baseRecalibrationKnownVariants.size()==0) {
+                throw new UserException.CommandLineException(NO_DBSNP_EXCEPTION);
+            }
+
+            // TODO: once we move to our own thing, make RECAL_TABLE_FILE optional
+            saveTextualTables = (null != BRAC.RAC.RECAL_TABLE_FILE);
+
+            String referenceURL = BRAC.referenceArguments.getReferenceFileName();
+
+            if (BRAC.readArguments.getReadFilesNames().size()!=1) {
+                throw new UserException("Sorry, we only support a single reads input for now.");
+            }
+            String bam = BRAC.readArguments.getReadFilesNames().get(0);
+
+            if (!BucketUtils.isCloudStorageUrl(bam) && isRemote()) {
+                throw new UserException("Sorry, for remote execution the BAM must be stored remotely.");
+            }
+
+            // Load the input bam
+            final ReadsDataflowSource readsDataflowSource = new ReadsDataflowSource(bam, pipeline);
+            readsHeader = readsDataflowSource.getHeader();
+            final SAMSequenceDictionary readsDictionary = readsHeader.getSequenceDictionary();
+            final List<SimpleInterval> intervals = BRAC.intervalArgumentCollection.intervalsSpecified() ? BRAC.intervalArgumentCollection.getIntervals(readsHeader.getSequenceDictionary())
+                : IntervalUtils.getAllIntervalsForReference(readsHeader.getSequenceDictionary());
+            final PCollectionView<SAMFileHeader> headerSingleton = ReadsDataflowSource.getHeaderView(pipeline, readsHeader);
+            CountingReadFilter filterToApply = BaseRecalibratorDataflow2.readFilter(readsHeader);
+
+            bunny.stepEnd("set up bam input");
+
+            // Load the Variants and the Reference
+            //final VariantsDataflowSource variantsDataflowSource = new VariantsDataflowSource(baseRecalibrationKnownVariants, pipeline);
+            final ReferenceDataflowSource referenceDataflowSource = new ReferenceDataflowSource(pipeline.getOptions(), referenceURL, BQSR_REFERENCE_WINDOW_FUNCTION);
+            bunny.stepEnd("create referenceDataflowSource");
+            final SAMSequenceDictionary refDictionary = referenceDataflowSource.getReferenceSequenceDictionary(readsDictionary);
+            bunny.stepEnd("load ref sequence dictionary");
+
+            checkSequenceDictionaries(refDictionary, readsDictionary);
+            bunny.stepEnd("checkSequenceDictionaries");
+            PCollectionView<SAMSequenceDictionary> refDictionaryView = pipeline.apply(Create.of(refDictionary)).setName("refDictionary").apply(View.asSingleton());
+            bunny.stepEnd("create ref dictionary view");
+
+            // the granularity at which we'll want to assign work and read inputs.
+            final int bigShardSize = 1_000_000;
+            // the granularity at which we want to batch reads,variants,and reference for processing.
+            final int smallShardSize = 5_000;
```

Comment:
Declare these constants at the top of the class instead of embedded within the code.